rules/xcode_sdk_frameworks.bzl

```diff
@@ -0,0 +1,446 @@
+"""Repository rule to fetch and configure SDK Clang and Swift modules for explicit module builds.
+
+In explicit module builds, Swift targets need to depend directly on SDK Clang and Swift modules.
+This repository rule is to configure the targets for SDK Clang and Swift modules.
+The high-level idea is to locates all local Xcode versions, scan through all SDK frameworks, and
+configure targets for them. 
+"""
+
+load("@bazel_tools//tools/osx:xcode_configure.bzl", "run_xcode_locator")
+
+OSX_EXECUTE_TIMEOUT = 600
+
+PLATFORM_TUPLES = [
+    ("MacOSX", "macos"),
+    ("iPhoneOS", "ios"),
+    ("iPhoneSimulator", "ios"),
+]
+
+SDK_BUILD_FILE_TMPL = """load("@build_bazel_rules_apple//apple:apple.bzl", "apple_dynamic_framework_import")
+load("@build_bazel_rules_ios//rules/explicit_module:sdk_clang_module.bzl", "sdk_clang_module")
+load("@build_bazel_rules_swift//swift:swift.bzl", "swift_module_alias")
+
+package(default_visibility = ["//visibility:public"])
+{sdk_targets}
+"""
+
+IMPORTS_FILE = "bazel_xcode_imports.swift"
+
+CLANG_MODULE_TMPL = """
+sdk_clang_module(
+    name = "{name}_c",
+    module_name = "{name}",
+    module_map = "{module_map_path}",
+    deps = [
+{deps}
+    ],
+)
+"""
+
+SWIFT_MODULE_TMPL = """
+swift_module_alias(
+    name = "{name}_swift",
+    module_name = "{name}",
+    deps = [
+{deps}
+    ],
+)
+"""
+
+ROOT_BUILD_FILE_TMPL = """load("@bazel_skylib//lib:selects.bzl", "selects")
+
+package(default_visibility = ["//visibility:public"])
+
+selects.config_setting_group(
+    name = "iPhoneOS",
+    match_any = [
+        "@build_bazel_rules_apple//apple:ios_arm64",
+        "@build_bazel_rules_apple//apple:ios_arm64e",
+    ],
+)
+
+selects.config_setting_group(
+    name = "iPhoneSimulator",
+    match_any = [
+        "@build_bazel_rules_apple//apple:ios_sim_arm64",
+        "@build_bazel_rules_apple//apple:ios_x86_64",
+    ],
+)
+
+selects.config_setting_group(
+    name = "MacOSX",
+    match_any = [
+        "@build_bazel_rules_apple//apple:darwin_arm64",
+        "@build_bazel_rules_apple//apple:darwin_arm64e",
+        "@build_bazel_rules_apple//apple:darwin_x86_64",
+    ],
+)
+{config_setting_lines}
+{xcode_sdk_framework_alias}
+"""
+
+ROOT_ALIAS_TMPL = """
+alias(
+    name = "{target}",
+    actual = select({{
```

Comment:
i got error here on 
```
ERROR: /private/var/tmp/_bazel_yuanfeng/8f12d9050b931ac3050ca4cf9fc59845/external/_main~xcode_sdk_frameworks~xcode_sdk_frameworks/BUILD.bazel:38:6: configurable attribute "actual" in @@_main~xcode_sdk_frameworks~xcode_sdk_frameworks//:xcode_sdk_frameworks doesn't match this configuration. Would a default condition help?

Conditions checked:
 @@_main~xcode_sdk_frameworks~xcode_sdk_frameworks//:xcode_15_4_0_15F31d

To see a condition's definition, run: bazel query --output=build <condition label>.
```
i tested by adding `"//conditions:default": "//version15_4_0_15F31d:xcode_sdk_frameworks",` here and the build now succeed. So why you don't hit it and it sounds like we need to ask for a default xcode version or sth?

Reply:
That's because the repo is pinned to 15.2 while you don't have 15.2 installed. You can update the xcode version in .bazelrc to `build --xcode_version=15.4.0.15F31d`